src/app/dim-ui/WeaponDeepsightInfo.tsx

```diff
@@ -9,17 +16,70 @@ import styles from './WeaponDeepsightInfo.m.scss';
 export function WeaponDeepsightInfo({ item }: { item: DimItem }) {
   const deepsightInfo = item.deepsightInfo;
   const record = item.patternUnlockRecord;
+  const relevantObjectives = record?.objectives.filter((o) => !o.complete);
 
-  if (!deepsightInfo && !record?.objectives[0]) {
+  if (!deepsightInfo && !relevantObjectives?.length) {
     return null;
   }
 
   return (
     <div className={styles.deepsightProgress}>
-      {deepsightInfo && <Objective objective={deepsightInfo.attunementObjective} />}
-      {record?.objectives.map((objective) => (
-        <Objective key={objective.objectiveHash} objective={objective} />
-      ))}
+      {deepsightInfo ? (
+        <>
+          <PatternUnlockedIndicator record={record} />
+          <div className={styles.deepsightProgressSection}>
+            <Objective objective={deepsightInfo.attunementObjective} />
+          </div>
+        </>
+      ) : (
+        Boolean(relevantObjectives?.length) && (
```

Comment:
Shouldn't need the `?.` here right?

Reply:
hmm, the check on line 21 is an &&, so one or the other could fail and it might be undefined still by this point